### PR TITLE
test: add e2e tests for `https` option

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case

add e2e tests for `https` option.

### Breaking Changes
No

### Additional Info
puppeteer having issues accepting SSL with `requestCert: true`, throwing error `net::ERR_BAD_SSL_CLIENT_AUTH_CERT`, hence testing with `supertest` for this case.

### CODE REVIEW

File: test/helpers/run-browser.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ function runBrowser(config) {
     puppeteer
       .launch({
         headless: true,
+        ignoreHTTPSErrors: true,
```
Comment (alexander-akait) [inline test/helpers/run-browser.js:23]: Due invalid localhost cert? Let's add comment about it

Comment (snitin315) [reply test/helpers/run-browser.js:24]: yes, done 👍🏻

**Review by alexander-akait — APPROVED** [review]
Feel free to merge when it will be green

**Comment by alexander-akait** [comment]
Note, coverage should be not changed when we rewrite tests, because we can missing something from old tests

**Comment by snitin315** [comment]
All green ✅

**Pull request merged**